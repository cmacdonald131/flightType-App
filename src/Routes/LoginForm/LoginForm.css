* {
  box-sizing: border-box;
}

.LoginForm_head {
  background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(../../Images/wing.jpg);
  background-position: center center;
  background-size: cover;
  color: #ffffff;
  height: 100vh;
}

.Login_section {
  text-align: center;
}

.label {
  font-size: 18px;
  margin-bottom: 5px;
  margin-right: 5px;
}

button {
  padding: 5px 10px;
  margin-top: 15px;
}

input {
  padding-top: 5px;
  padding-bottom: 5px;
  border: 1px solid black;
}

input::placeholder {
  padding-left: 5px;
  font-size: 16px;
}

@media screen and (min-width: 700px) {
  .LoginForm_head {
    background: linear-gradient(rgba(0, 0, 0, 0.3), rgba(0, 0, 0, 0.3)), url(../../Images/flight.jpg);
    background-position: center center;
    background-size: cover;
    color: #ffffff;
    height: 100vh;
  }

  .Login_section {
      width: 60%;
      margin-right: auto;
      margin-left: auto;
  }
}
