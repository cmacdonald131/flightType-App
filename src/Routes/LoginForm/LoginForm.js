import React, { Component } from 'react'
import Navbar from '../../Components/Navbar/Navbar'
import AuthApiService from '../../Services/auth-api-service'
import ApiContext from '../../ApiContext'
import './LoginForm.css'

export default class LoginForm extends Component {
    static contextType = ApiContext
    static defaultProps = {
    }

    //when a successful login occurs the team page is rendered

    state = { error: null }
    onLoginSuccess = () => {
        this.props.history.push('/user')
    }

    //On clicking the Login button when entering correct login information a call is sent to post an authorized login to the server through a JWT 
    handleSubmitJwtAuth = ev => {
        ev.preventDefault()
        this.setState({ error: null })
        const { username, password } = ev.target

        AuthApiService.postLogin({
            username: username.value,
            password: password.value,
        })
            .then(res => {
                this.context.setUser(username.value)
                username.value = ''
                password.value = ''
                this.onLoginSuccess()
            })
            .catch(res => {
                this.setState({ error: res.error })
                alert("We couldn't log you in with the username and/or password you entered.  Please try again.")
            })
    }
    render() {
        return (
            <div className="LoginForm_head">
                <form
                    className='LoginForm'
                    onSubmit={this.handleSubmitJwtAuth}
                >
                    <Navbar />
                    <header>
                        <h1 style={{ textAlign: "center" }}>Login to your account</h1>
                    </header>
                    <section className="Login_section">
                        <div className='username' style={{ marginBottom: "10px" }}>
                            <label htmlFor='LoginForm__username' className="label" style={{ textAlign: "left" }}>
                                Username
                            </label>
                            <input
                                required
                                name='username'
                                placeholder='Enter Username'
                                id='LoginForm__username'
                                aria-label="Enter your username"
                            >
                            </input>
                        </div>
                        <div className='password' style={{ marginBottom: "10px" }}>
                            <label htmlFor='LoginForm__password' className="label" style={{ textAlign: "left" }}>
                                Password
                            </label>
                            <input
                                required
                                name='password'
                                type='password'
                                placeholder='Enter Password'
                                id='LoginForm__password'
                                aria-label="Enter your password"
                            >
                            </input>
                        </div>
                        <button
                            className="submit"
                            type='submit'
                            aria-label="Login"
                        >Login
                        </button>
                    </section>
                </form>
            </div>
        )
    }
}