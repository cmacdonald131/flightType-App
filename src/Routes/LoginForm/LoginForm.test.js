import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LoginForm from './LoginForm'

describe(`LoginForm component`, () => {
    const props = {
        username: 'test-username',
        password: 'test-password',
        'data-other': 'test-other-prop'
    }

    it('renders a form.LoginForm by default', () => {
        const wrapper = shallow(<LoginForm />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    it('renders the LoginForm given props', () => {
        const wrapper = shallow(<LoginForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})