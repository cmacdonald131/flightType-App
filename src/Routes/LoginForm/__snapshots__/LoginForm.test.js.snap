// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component renders a form.LoginForm by default 1`] = `
<div
  className="LoginForm_head"
>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <Navbar />
    <header>
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Login to your account
      </h1>
    </header>
    <section
      className="Login_section"
    >
      <div
        className="username"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <label
          className="label"
          htmlFor="LoginForm__username"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          Username
        </label>
        <input
          id="LoginForm__username"
          name="username"
          placeholder="Enter Username"
          required={true}
        />
      </div>
      <div
        className="password"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <label
          className="label"
          htmlFor="LoginForm__password"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          Password
        </label>
        <input
          id="LoginForm__password"
          name="password"
          placeholder="Enter Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="submit"
        type="submit"
      >
        Login
      </button>
    </section>
  </form>
</div>
`;

exports[`LoginForm component renders the LoginForm given props 1`] = `
<div
  className="LoginForm_head"
>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <Navbar />
    <header>
      <h1
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        Login to your account
      </h1>
    </header>
    <section
      className="Login_section"
    >
      <div
        className="username"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <label
          className="label"
          htmlFor="LoginForm__username"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          Username
        </label>
        <input
          id="LoginForm__username"
          name="username"
          placeholder="Enter Username"
          required={true}
        />
      </div>
      <div
        className="password"
        style={
          Object {
            "marginBottom": "10px",
          }
        }
      >
        <label
          className="label"
          htmlFor="LoginForm__password"
          style={
            Object {
              "textAlign": "left",
            }
          }
        >
          Password
        </label>
        <input
          id="LoginForm__password"
          name="password"
          placeholder="Enter Password"
          required={true}
          type="password"
        />
      </div>
      <button
        className="submit"
        type="submit"
      >
        Login
      </button>
    </section>
  </form>
</div>
`;
